### PR TITLE
feat: mobile layout and XEP detection

### PR DESCRIPTION
**Pull request by misterupkeep**
This PR introduces several breaking changes:
 - Infinite scrolling is removed
 - Message archiving and sync is questionably functional
 - There is no search UI on mobile
 - XEP-0352 is missing in the 'features' event in stanza.js, meaning this XEP will never be detected

### CODE REVIEW

**Comment by misterupkeep** [comment]
Pulling should open 4 issues - one for each breaking change specified above

File: store/xeps.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+const $getters = {
+    getServerXep: "GET_SERVER_XEP",
+};
+
+const $mutations = {
+    setServerXep: "SET_SERVER_XEP",
+}
+
+export const state = () => ({
```
Comment (00drip) [inline store/xeps.js:17]: We should probably add comments here describing every XEP/RFC, e.g. 
```js
"XEP-0030": false, // service discovery
```

File: store/search.js [hunk 2]
```diff
@@ -1,27 +1,30 @@
 const FuzzyMatching = require("fuzzy-matching");
 
-const kWordBreakRegex = new RegExp(/\P{L}+/u);
```
Comment (00drip) [inline store/search.js:3]: Why? `\P{L}` is unicode non-letter chars, do we want `a.b.c` to be itself or be `a`, `b`, and `c`?

File: components/Chat/Message/message.scss [hunk 3]
```diff
@@ -22,20 +22,6 @@
   overflow: hidden;
 }
 
-.highlight {
```
Comment (00drip) [inline components/Chat/Message/message.scss:25]: Why get rid of this? Is it replaced elsewhere?

File: components/Chat/MessageGroup.vue [hunk 4]
```diff
@@ -1,10 +1,8 @@
 <template functional>
   <div class="mb-2">
-    <div v-for="(message, i) in props.group" :key="i">
-      <span :id="`msg:${message.timestamp}`"/>
-      <message :heading="i === 0" darken-on-hover
-        :from="message.from" :body="message.body" :timestamp="message.timestamp"/>
-    </div>
+    <message v-for="(message, i) in props.group" :key="i"
```
Comment (00drip) [inline components/Chat/MessageGroup.vue:3]: How would we access this? I was having problems with dynamic refs not showing up on `this.$refs`.

File: store/xeps.js (hunk 1 above)
Comment (misterupkeep) [reply store/xeps.js:17]: You have an explanation in [`/assets/xeps/server/data.js`](https://github.com/bonfire-xmpp/bonfire/blob/feature/mobile-layout/assets/xeps/server/data.js) locally, and running an online search should help. Sadly, such is the way of the XEP. And I don't feel like rewriting tons of boilerplate comments.

File: components/Settings/Panels/XEPs.vue [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    supportedNo() { return this.xeps.reduce((acc, x) => x.supported ? acc+1 : acc, 0) },
+    supportedPct() { return Math.round(100 * this.supportedNo / this.xeps.length) },
+  },
+
+  methods: {
+    ...mapActions(XEPStore.namespace, {
+      refreshXeps: XEPStore.$actions.updateXepsWithDisco,
+    }),
+
+    async refresh() {
```
Comment (00drip) [inline components/Settings/Panels/XEPs.vue:87]: I think the flickering should only last as long as the query is active, then the 'waterfall' delay effect should play.

File: store/search.js (hunk 2 above)
Comment (misterupkeep) [reply store/search.js:3]: See: fffa60e2032e00d385af0816ce9e29276143b9dc 
```
chore: revert "feat(scrollback): add infinite message scrollback"

This reverts commit 862c1b0, reversing
changes made to 0bf5c20.
```
This commit made things too buggy, crashed a lot, and was hard to develop with.

File: components/Chat/Message/message.scss (hunk 3 above)
Comment (misterupkeep) [reply components/Chat/Message/message.scss:25]: See: fffa60e2032e00d385af0816ce9e29276143b9dc 
```
chore: revert "feat(scrollback): add infinite message scrollback"

This reverts commit 862c1b0, reversing
changes made to 0bf5c20.
```
This commit made things too buggy, crashed a lot, and was hard to develop with.

File: components/Chat/MessageGroup.vue (hunk 4 above)
Comment (misterupkeep) [reply components/Chat/MessageGroup.vue:3]: See: fffa60e2032e00d385af0816ce9e29276143b9dc 
```
chore: revert "feat(scrollback): add infinite message scrollback"

This reverts commit 862c1b0, reversing
changes made to 0bf5c20.
```
This commit made things too buggy, crashed a lot, and was hard to develop with.

**Review by 00drip — COMMENTED** [review]
Good to merge, just look over the comments.

File: store/search.js (hunk 2 above)
Comment (00drip) [reply store/search.js:3]: Sounds good, we could add that back eventually.

**Pull request merged**